### DISCUSSION TITLE
Commerce module in OrchardCore main project - PackageReference vs. ProjectReference

### DISCUSSION THREAD

**Discussion by havana7** [OrchardCMS/OrchardCore.Commerce#282] [Q&A] [answered]
I added the OrchardCore.Commerce project as a module to the original OrchardCore project.

If I leave the PackageReferences I can see and activate the Commerce module in the Features section, but if I delete the PackageReferences and add all the individual ProjectReferences the module is not displayed.

Commented as PackageReference 

``

Added as ProjectReference 

`		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore.Modules\OrchardCore.ContentFields\OrchardCore.ContentFields.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore.Modules\OrchardCore.ContentTypes\OrchardCore.ContentTypes.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore.Modules\OrchardCore.Html\OrchardCore.Html.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore.Modules\OrchardCore.Templates\OrchardCore.Templates.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore.Modules\OrchardCore.Workflows\OrchardCore.Workflows.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.Admin.Abstractions\OrchardCore.Admin.Abstractions.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement.Abstractions\OrchardCore.ContentManagement.Abstractions.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.ContentManagement\OrchardCore.ContentManagement.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.ContentTypes.Abstractions\OrchardCore.ContentTypes.Abstractions.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.Workflows.Abstractions\OrchardCore.Workflows.Abstractions.csproj" />
		<ProjectReference Include="..\..\..\..\OrchardCore\src\OrchardCore\OrchardCore.Module.Targets\OrchardCore.Module.Targets.csproj" />
`

My goal is to have the Commerce module source in the OrchardCore main project source.

Is it a feasible solution?

Thank you

**Comment by sarahelsaig** [comment, marked as answer]
This is not a supported scenario. Some of the other packages (e.g. Lombiq.HelpfulLibraries.OrchardCore) also reference the OC packages; so both will end up loaded which causes conflicts - unless you include a lot of altered source in your repo. Generally it's not a good developer practice to include the OC source code because it slows down development and compilation. Why do you want it?

**Comment by havana7** [comment]
I wanted to have a single source for all code so that I could easily navigate through all projects, classes and implementations.

The build would only happen once and after that I would disable all already built projects keeping only the ones I work on.

Thank you for your answer.
